src/RestApi/StoreApi/Controllers/CartOrder.php

```diff
@@ -0,0 +1,362 @@
+		$order->set_prices_include_tax( 'yes' === get_option( 'woocommerce_prices_include_tax' ) );
+		$order->set_customer_id( get_current_user_id() );
+		$order->set_customer_ip_address( \WC_Geolocation::get_ip_address() );
+		$order->set_customer_user_agent( wc_get_user_agent() );
+
+		$this->set_props_from_cart( $order, $request );
+		$this->create_line_items_from_cart( $order, $request );
+		$this->set_props_from_request( $order, $request );
+
+		$order->save();
+
+		// Store Order ID in session so we can look it up later.
+		WC()->session->set( 'draft_order_id', $order->get_id() );
+
+		return $order;
+	}
+
+	/**
+	 * Get an order object, either using a current draft order, or returning a new one.
+	 *
+	 * @return \WC_Order A new order object.
+	 */
+	protected function get_order_object() {
+		$draft_order_id = WC()->session->get( 'draft_order_id' );
+		$draft_order    = $draft_order_id ? wc_get_order( $draft_order_id ) : false;
+
+		if ( $draft_order && $draft_order->has_status( 'draft' ) && 'store-api' === $draft_order->get_created_via() ) {
+			return $draft_order;
+		}
+
+		return new \WC_Order();
+	}
+
+	/**
+	 * Changes default order status to draft for orders created via this API.
+	 *
+	 * @return string
+	 */
+	public function default_order_status() {
+		return 'draft';
+	}
+
+	/**
+	 * Set order totals based on those calculated by the cart.
+	 *
+	 * @param \WC_Order   $order Object to prepare for the response.
+	 * @param RestRequest $request Full details about the request.
+	 */
+	protected function set_props_from_cart( &$order, $request ) {
+		$order->set_shipping_total( WC()->cart->get_shipping_total() );
+		$order->set_discount_total( WC()->cart->get_discount_total() );
+		$order->set_discount_tax( WC()->cart->get_discount_tax() );
+		$order->set_cart_tax( WC()->cart->get_cart_contents_tax() + WC()->cart->get_fee_tax() );
+		$order->set_shipping_tax( WC()->cart->get_shipping_tax() );
+		$order->set_total( WC()->cart->get_total( 'edit' ) );
+		$order->update_meta_data( 'is_vat_exempt', WC()->cart->get_customer()->get_is_vat_exempt() ? 'yes' : 'no' );
+	}
+
+	/**
+	 * Create order line items.
+	 *
+	 * @param \WC_Order   $order Object to prepare for the response.
+	 * @param RestRequest $request Full details about the request.
+	 */
+	protected function create_line_items_from_cart( &$order, $request ) {
+		$draft_order_cart_hash = $order->get_cart_hash();
+
+		if ( WC()->cart->get_cart_hash() === $draft_order_cart_hash ) {
+			return;
+		}
+
+		if ( $draft_order_cart_hash ) {
+			$order->remove_order_items();
+		}
+
+		WC()->checkout->create_order_line_items( $order, WC()->cart );
+		WC()->checkout->create_order_fee_lines( $order, WC()->cart );
+		// @todo This may need revision during checkout implementation in order to store chosen shipping options rather than those from the cart.
+		WC()->checkout->create_order_shipping_lines( $order, WC()->session->get( 'chosen_shipping_methods' ), WC()->shipping()->get_packages() );
```

Comment:
Question: since it appears it's possible to update the existing draft order and it will pull the latest from the cart, can we effectively update the order too by posting changes to the cart endpoint before posting to `cart/order`?  If so, that's likely how it can be implemented when this we start wiring up the checkout.

Reply:
Yes we can. I was thinking about this some more and I think we can essentially "lock" the order contents as the user enters checkout, pulling details from the draft order instead of the cart. Still probably some details to flesh out, but yes to your question.